tests_v2/unit/app/routes/datasets/test_asset_metadata.py

```diff
@@ -42,12 +50,41 @@ async def test_update_asset_metadata(
     (tile_cache_asset,) = assets
     assert tile_cache_asset.metadata.max_zoom == max_zoom_before_update
 
-    resp = await async_client.patch(
-        f"asset/{tile_cache_asset.asset_id}/metadata", json={"max_zoom": max_zoom}
-    )
-
-    assert resp.json()["status"] == "success"
-    assert resp.json()["data"]["max_zoom"] == max_zoom
+    # generic_vector_source_version creates with owner of manager_mocked, so
+    # update by manager_mocked should succeed
+    appmain.dependency_overrides[get_user] = get_manager_mocked
+
```

Comment:
Not necessarily in this PR, but wondering if there's a way for us to turn this into a reusable helper function since we keep copying this code block